backend/DataAccess/Configuration/LfIdSerializer.cs

```diff
@@ -42,15 +53,42 @@ public override JsonConverter CreateConverter(Type typeToConvert, JsonSerializer
     {
         return new LfIdSerializer(typeToConvert);
     }
+
+    private static bool IsLfId(Type type)
+    {
+        return type.IsAssignableTo(typeof(LfId));
+    }
+
+    private static bool IsDictionaryOfLfIds(Type type)
+    {
+        return type.GetInterfaces().Any(i => i.IsGenericType &&
+            (i.GetGenericTypeDefinition() == typeof(IDictionary<,>)) &&
+            IsLfId(i.GetGenericArguments()[0]));
+    }
+
+    private static IBsonSerializer CreateDictionarySerializer<TKey, TValue>() where TKey : notnull
```

Comment:
nice job, still complicated but at least it's more readable then using reflection.